### PR TITLE
Fix possible Windows GPU multiple window crash

### LINKED ISSUE #4000: Develop branch: Crash with multiple windows
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

I was working with the develop branch and had some crazy crashes going on with my Windows build. I finally trimmed it down to this minimal example.

### How to reproduce

1. On Ubuntu 22.04.2 LTS, go version go1.20.4 linux/amd64, cross compiling to Windows 11, x86_64-w64-mingw32-gcc (GCC) 10-win32 20220113
2. mkdir fresh
3. cd fresh
4. go mod init testui
5. git clone https://github.com/fyne-io/fyne
6. cd fyne
7. git checkout develop
8. cd ..
9. go mod edit -replace=fyne.io/fyne/v2=./fyne
10. mkdir testui
11. Place attached example code in testui/main.go
12. cp fyne/theme/icons/fyne.png testui/Icon.png
13. go mod tidy
14. cd testui
15. CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc fyne package --tags fyne --os windows
16. Run the resulting testui.exe on Windows and see it crash -- sometimes immediately, sometimes you have to move the mouse over the button to trigger the crash.

### Screenshots

_No response_

### Example code

```go
package main

import (
        "fyne.io/fyne/v2/app"
        "fyne.io/fyne/v2/widget"
)

func main() {
        application := app.NewWithID("testui")
        window1 := application.NewWindow("Window One")
        window1.SetContent(widget.NewButton("Button", nil))
        window2 := application.NewWindow("Window Two")
        window1.Show()
        window2.Show()
        application.Run()
}
```

### Fyne version

develop branch

### Go compiler version

1.20.4

### Operating system and version

Compiling on Ubuntu 22.04.2 LTS, running on Windows 11

### Additional Information

I'm working to get any stderr output, but that seems like it's own fun game with Windows.
I will post it once I get it.

### PR DESCRIPTION
**Pull request by andydotxyz**
This was a real pain!
On some GPUs the new shaders would crash when in the background.
It appears to be like the discovered webGL issue before but worse.
This fix addresses both at the same time.

Tested in Windows Intel, webGL and also GLES code paths.

Fixes #4000

### Checklist:

- [ ] Tests included. <- see the issue for steps (you'll need the right GPU)
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by Goltanju** [comment]
:+1: Works great here. Verified develop failed, then that fix/4000 passed. nVidia GeForce RTX 2080 Ti on Windows 11.

That looks like it was "fun". Thankyou!

File: cmd/hello/FyneApp.toml
Comment (Jacalz) [inline cmd/hello/FyneApp.toml:1]: Was this change intentional?

**Review by Jacalz — COMMENTED** [review]
While I haven't been able to reproduce the issue myself, the code does look good and works fine on my computer. Well done!

Just had one not inline about a file that probably shouldn't be deleted. Consider it approved after that :)

File: cmd/hello/FyneApp.toml
Comment (andydotxyz) [reply cmd/hello/FyneApp.toml:1]: Oops I wonder how that happened sorry!

**Review by Bluebugs — DISMISSED** [review]
I have some swear word I want to use about GL driver. Good catch. Thanks for tracking this down.

File: cmd/hello/FyneApp.toml
Comment (andydotxyz) [reply cmd/hello/FyneApp.toml:1]: Fixed, sorry about that

**Review by Jacalz — APPROVED** [review]

File: cmd/hello/FyneApp.toml
Comment (Jacalz) [reply cmd/hello/FyneApp.toml:1]: No problem. We all make mistakes :)
